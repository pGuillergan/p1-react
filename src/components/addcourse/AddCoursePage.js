import React, { useState } from "react";
import { toast } from "react-toastify";
import AddCourseForm from "./AddCourseForm";

export default function AddCoursePage(props) {
  const [course, setCourse] = useState({
    course_title: "",
    courseDescription: "",
    units: "",
  });

  function handleChange(event) {
    setCourse({ ...course, [event.target.name]: event.target.value });
  }

  function handleSubmit(event) {
    event.preventDefault();
    fetch("http://54.215.227.33:3000/addcourse", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(course),
    })
      .then((result) => {
        toast.success("Added new course!");
        props.history.push("/courses");
      })
      .catch((err) => {
        toast.error("Course was not added..");
      });
  }

  return (
    <div>
    <div className="jumbotron">
      <h3>Add new course</h3>
      <AddCourseForm
        onSubmit={handleSubmit}
        onChange={handleChange}
        course={course}
      />
      </div>
    </div>
  );
}
