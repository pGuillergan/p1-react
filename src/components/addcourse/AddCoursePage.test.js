import React, { Component } from "react";
import { shallow } from "enzyme";

import AddCoursePage from "./AddCoursePage";

describe("AddPage", () => {
  it("should add courses", () => {
    const addPage = shallow(<AddCoursePage/>);
    expect(addPage).toMatchSnapshot();

    expect(addPage.find('h3').length).toEqual(1);
    // expect(addPage.find('AddCourseForm').length).toEqual(1);

  });

});