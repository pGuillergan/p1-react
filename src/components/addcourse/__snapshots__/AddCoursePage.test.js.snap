// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPage should add courses 1`] = `
<div>
  <h3>
    Add new course
  </h3>
  <CreateAddCourseForm
    course={
      Object {
        "courseDescription": "",
        "course_title": "",
        "units": "",
      }
    }
    onChange={[Function]}
    onSubmit={[Function]}
  />
</div>
`;
